Sources/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift

```diff
@@ -0,0 +1,324 @@
+
+import CTensorFlow
+
+/// Swift-side convenience wrapper for a `TF_Graph`. It holds a pointer to the
+/// underlying TF_Graph and also exposes some aspects of the TF_Graph as
+/// properties.
+class TFGraph {
+    /// The `TF_Operation *` type.
+    typealias CTFOperation = OpaquePointer
+
+    enum Input {
+        case single(TF_Output)
+        case list([TF_Output])
+    }
+
+    /// The pointer to the underlying TF Graph.
+    let cTFGraph: CTFGraph = TF_NewGraph()
+    var inputs: [TF_Output] = []
+    var nodes: [CTFOperation?] = []
+    var outputs: [TF_Output] = []
+    /// An array representing how the outputs are grouped. The grouping
+    /// corresponds to a higher-level notion like TensorGroup.
+    var outputGroupCounts: [Int] = []
+    var name: String { "lazyTrace_\(nodeCounter)" }
+
+    /// A status object to pass to TF graph building operations.
+    private let status: CTFStatus = TF_NewStatus()
+
+    /// Counter that is used for number the generated graph nodes.
+    private var nodeCounter: Int = 0
+
+    init(_ lazyTrace: LazyTensorTrace) {
+        var nodesCache: [ObjectIdentifier: CTFOperation?] = [:]
+        for op in lazyTrace.operations {
+            let opInputs = op.inputs.map { input -> TFGraph.Input in
+                switch input {
+                case LazyTensorOperation.Input.single(let h):
+                    return TFGraph.Input.single(makeTFOutput(h, nodesCache))
+                case LazyTensorOperation.Input.list(let elements):
+                    let tfInputs = elements.map { makeTFOutput($0, nodesCache) }
+                    return TFGraph.Input.list(tfInputs)
+                }
+            }
+            let graphNode = makeTFGraphNode(
+                name: op.name,
+                attrs: op.attributes,
+                inputs: opInputs,
+                device: op.deviceName)
+            nodesCache[ObjectIdentifier(op)] = graphNode
+            if op.name != "Placeholder" { nodes.append(graphNode) }
+        }
+        self.inputs = lazyTrace.inputs.map {
+            TF_Output(oper: nodesCache[ObjectIdentifier($0)]!, index: 0)
+        }
+        for output in lazyTrace.outputs {
+            let graphNode = nodesCache[ObjectIdentifier(output)]!
+            outputGroupCounts.append(output.outputCount)
+            outputs += (0..<output.outputCount).map {
+                TF_Output(oper: graphNode, index: Int32($0))
+            }
+        }
+    }
+
+    deinit {
+        TF_DeleteGraph(cTFGraph)
+        TF_DeleteStatus(status)
+    }
+
+    private func newNodeName(base: String) -> String {
+        let name = "\(base)_\(nodeCounter)"
+        nodeCounter += 1
+        return name
+    }
+
+    private func updateAttribute(
+        _ desc: CTFOperationDescription,
+        _ name: String,
+        _ attrValue: LazyTensorOperation.Attribute
+    ) {
+        switch attrValue {
+        case LazyTensorOperation.Attribute.tensorDataTypeValue(let value):
```

Comment:
```suggestion
        case .tensorDataTypeValue(let value):
```
Same for all other occurrences.